packages/common-ui-web/test/components/TakePictureButton.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { TakePictureButton } from '../../src';
+
+const TAKE_PICTURE_BTN_TEST_ID = 'take-picture-btn';
+const TAKE_PICTURE_OUTER_BTN_TEST_ID = 'take-picture-outer-btn';
+
+describe('TakePictureButton component', () => {
+  it('should take the size prop into account', () => {
+    const size = 60;
+    const borderWidth = 3;
+    const margin = 3;
+    const { unmount } = render(<TakePictureButton disabled={false} size={size} />);
+
+    const outerDiv = screen.getByTestId(TAKE_PICTURE_OUTER_BTN_TEST_ID);
+    const innerDiv = screen.getByTestId(TAKE_PICTURE_BTN_TEST_ID);
+    expect(outerDiv.style.width).toEqual(`${size - (2 * borderWidth)}px`);
+    expect(outerDiv.style.height).toEqual(`${size - (2 * borderWidth)}px`);
+    expect(innerDiv.style.width).toEqual(`${size - (2 * borderWidth) - (2 * margin)}px`);
+    expect(innerDiv.style.height).toEqual(`${size - (2 * borderWidth) - (2 * margin)}px`);
+    unmount();
+  });
+
+  it('should display a take picture button that calls the handle.takePicture method', () => {
+    const handle = { takePicture: jest.fn() };
+    const { unmount } = render(<TakePictureButton disabled={false} onClick={handle?.takePicture} size={60} />);
+
+    fireEvent.click(screen.getByTestId(TAKE_PICTURE_BTN_TEST_ID));
+    expect(handle.takePicture).toHaveBeenCalled();
+    unmount();
+  });
+
+  it('should not throw any error when no handle is provided', () => {
+    expect(() => {
+      const { unmount, rerender } = render(<TakePictureButton disabled={false} size={60} />);
+      fireEvent.click(screen.getByTestId(TAKE_PICTURE_BTN_TEST_ID));
+
+      rerender(<TakePictureButton disabled={false} onClick={() => { }} size={60} />);
+      fireEvent.click(screen.getByTestId(TAKE_PICTURE_BTN_TEST_ID));
+      unmount();
+    }).not.toThrow();
+  });
+});
```

Comment:
After you will have implemented the changes I mentioned (the forward ref etc.), you will be missing the following tests :

```typescript jsx
it('should pass the disabled prop down to the button element', () => {
  expectComponentToPassDownOtherPropsToHTMLElement(TakePictureButton, TAKE_PICTURE_BTN_TEST_ID, { disabled: true });
});

it('should pass the className down to the button element', () => {
  expectComponentToPassDownClassNameToHTMLElement(TakePictureButton, TAKE_PICTURE_BTN_TEST_ID);
});

it('should pass the style down to the button element', () => {
  expectComponentToPassDownStyleToHTMLElement(TakePictureButton, TAKE_PICTURE_BTN_TEST_ID);
});

it('should pass other props down to the button element', () => {
  expectComponentToPassDownOtherPropsToHTMLElement(TakePictureButton, TAKE_PICTURE_BTN_TEST_ID);
});

it('should pass the ref down to the button element', () => {
  expectComponentToPassDownRefToHTMLElement(TakePictureButton, TAKE_PICTURE_BTN_TEST_ID);
});
```